File: Python3/1 - 200/167 Two Sum II - Input array is sorted.py
# 40ms, 100.0%
class Solution:
    def twoSum(self, numbers, target):
        rem = {}
        for i in range(len(numbers)):
            if numbers[i] in rem: return [rem[numbers[i]] + 1, i + 1]
            rem[target - numbers[i]] = i
    

File: 1 - 200/167 Two Sum II - Input array is sorted.cc
// 7ms, 18.16%
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        int beg = 0, end = numbers.size() - 1;
        while (true) {
            if (numbers[beg] + numbers[end] < target)
                ++beg;
            else if (numbers[beg] + numbers[end] > target)
                --end;
            else
                return vector<int>{beg + 1, end + 1};
        }
    }
};
